src/bokeh/util/sampledata.py

```diff
@@ -56,39 +56,48 @@ def download(progress: bool = True) -> None:
 
     # HTTP requests are cheaper for us, and there is nothing private to protect
     s3 = 'http://sampledata.bokeh.org'
-    files = json.load(open(Path(__file__).parent / "sampledata.json"))
+    files = metadata()
 
-    for filename, md5 in files:
-        real_name, ext = splitext(filename)
-        if ext == '.zip':
-            if not splitext(real_name)[1]:
-                real_name += ".csv"
-        else:
-            real_name += ext
-        real_path = data_dir /real_name
+    for file_name, md5 in files.items():
+        real_path = data_dir / real_name(file_name)
 
         if real_path.exists():
-            local_md5 = hashlib.md5(open(real_path, 'rb').read()).hexdigest()
+            with open(real_path, "rb") as file:
+                data = file.read()
+            local_md5 = hashlib.md5(data).hexdigest()
             if local_md5 == md5:
-                print(f"Skipping {filename!r} (checksum match)")
+                print(f"Skipping {file_name!r} (checksum match)")
                 continue
             else:
-                print(f"Re-fetching {filename!r} (checksum mismatch)")
+                print(f"Re-fetching {file_name!r} (checksum mismatch)")
 
-        _download_file(s3, filename, data_dir, progress=progress)
+        _download_file(s3, file_name, data_dir, progress=progress)
 
 #-----------------------------------------------------------------------------
 # Dev API
 #-----------------------------------------------------------------------------
 
+def real_name(name: str) -> str:
+    real_name, ext = splitext(name)
+    if ext == ".zip":
+        if not splitext(real_name)[1]:
+            return f"{real_name}.csv"
+        else:
+            return real_name
+    else:
+        return name
+
+def metadata() -> dict[str, str]:
+    return dict(json.load(open(Path(__file__).parent / "sampledata.json")))
+
 def external_csv(module: str, name: str, **kw: Any) -> pd.DataFrame:
     '''
 
     '''
```

Comment:
As long as this file is getting attention, these empty doctrings could get a single sentence